dash-renderer/src/TreeContainer.js

```diff
@@ -94,6 +94,60 @@ CheckedComponent.propTypes = {
     id: PropTypes.string,
 };
 class TreeContainer extends Component {
+    constructor(props) {
+        super(props);
+
+        this.setProps = this.setProps.bind(this);
+    }
+
+    setProps(newProps) {
```

Comment:
We were creating new `setProps` on each TreeContainer render. Everything in here is dependent on the injected props, it can be reused if the instance is reused. Should reduce the number of re-renders, if nothing else.

Reply:
:tada: